test/default-split-pane-tests.js

```diff
@@ -215,87 +377,155 @@ describe('Drag resizer', () => {
 
     it('2 divs', () => {
       const jsx = (
-        <SplitPane split="horizontal">
+        <SplitPane split='horizontal'>
           <div>one</div>
           <div>two</div>
         </SplitPane>
       );
 
-      asserter(jsx, {height: 601})
-        .assertRatios([50, 50])
-        .assertSizes([300, 300])
+      asserter(jsx, {height: 600})
+        .assertSizes(['1', '1'])
+        .assertSizesPx([300, 300])
         .dragResizer(0, {y: -100})
-        .assertSizes([200, 400])
-        .assertRatios([33, 67])
+        .assertSizes(['20000', '40000'])
+        .assertSizesPx([200, 400])
       ;
     });
 
     it('2 Panes', () => {
       const jsx = (
-        <SplitPane split="horizontal">
+        <SplitPane split='horizontal'>
           <Pane>one</Pane>
           <Pane>two</Pane>
         </SplitPane>
       );
 
-      asserter(jsx, {height: 601})
-        .assertRatios([50, 50])
-        .assertSizes([300, 300])
+      asserter(jsx, {height: 600})
+        .assertSizes(['1', '1'])
+        .assertSizesPx([300, 300])
         .dragResizer(0, {y: -100})
-        .assertSizes([200, 400])
-        .assertRatios([33, 67])
+        .assertSizes(['20000', '40000'])
+        .assertSizesPx([200, 400])
+      ;
+    });
+
+    it('2 Panes, should persists initial size unit, px', () => {
+      const jsx = (
+        <SplitPane split='horizontal'>
+          <Pane initialSize='200px'>one</Pane>
+          <Pane>two</Pane>
+        </SplitPane>
+      );
+
+      asserter(jsx, {height: 600})
+        .assertSizes(['200px', '1'])
+        .assertSizesPx([200, 400])
+        .dragResizer(0, {y: 100})
+        .assertSizes(['300.00px', '1'])
+        .assertSizesPx([300, 300])
+      ;
+    });
+
+    it('2 Panes, should persists initial size unit, %', () => {
+      const jsx = (
+        <SplitPane split='horizontal'>
+          <Pane initialSize='30%'>one</Pane>
+          <Pane>two</Pane>
+        </SplitPane>
+      );
+
+      asserter(jsx, {height: 600})
+        .assertSizes(['30%', '1'])
+        .assertSizesPx([180, 420])
+        .dragResizer(0, {y: 120})
+        .assertSizes(['50.00%', '1'])
+        .assertSizesPx([300, 300])
       ;
     });
 
     it('2 Panes - resized all the way', () => {
       const jsx = (
-        <SplitPane split="horizontal">
+        <SplitPane split='horizontal'>
           <div>one</div>
           <div>two</div>
         </SplitPane>
       );
 
-      asserter(jsx, {height: 601})
-        .assertRatios([50, 50])
-        .assertSizes([300, 300])
+      asserter(jsx, {height: 600})
+        .assertSizesPx([300, 300])
+        .assertSizes(['1', '1'])
         .dragResizer(0, {y: 300})
-        .assertSizes([600, 0])
-        .assertRatios([100, 0])
+        .assertSizes(['60000', '0'])
```

Comment:
should this be `600` not `60000` ?

Reply:
Yes, it might be. I just multiply it by 100 to avoid float number. But I'm not sure that it somehow affects performance.
As I need to persist size units after resizing I convert the sizes from px to initial units. Px to px is clear, px to % - just calculates % of SplitPane container. For ratio just converts all sizes in ratio to px and multiply by 100.